contracts/misc/src/bridge/Bridge.sol

```diff
@@ -220,13 +220,11 @@ contract Bridge is Initializable, AccessControlUpgradeable, PausableUpgradeable,
         // Retrieve tokens from the sender, and deposit them into the lockbox for wrapping if necessary.
         if (wrap) {
             ILockbox(_lockbox).token().safeTransferFrom(msg.sender, address(this), value);
-            ILockbox(_lockbox).deposit(value);
-        } else {
-            _token.safeTransferFrom(msg.sender, address(this), value);
+            ILockbox(_lockbox).depositTo(msg.sender, value);
         }
 
         // Burn the tokens.
-        ITokenOps(_token).burn(value);
+        ITokenOps(_token).clawback(msg.sender, value);
```

Comment:
@Zodomo this was main change.

- using clawback means less token approvals needed
- RLUSD only has [clawbacker in initializer](https://github.com/ripple/RLUSD-Implementation/blob/07a27d74863a22f6ad403b5a9f8bb3aea76e05eb/contracts/StablecoinUpgradeable.sol#L44), so this way to we don't need another setup tx to assign BURNER role